test/EndToEndTests/Tests/Core/Microsoft.OData.Core.E2E.Tests/WriteJsonPayloadTests/SerializationInfoEdgeTests.cs

```diff
@@ -0,0 +1,488 @@
+                    Assert.Contains("$metadata#Parsen\"", result);
+                    Assert.Contains("People(-5)\",\"PersonId\":-5", result);
+                }
+                else
+                {
+                    Assert.DoesNotContain("People(-5)\",\"PersonId\":-5", result);
+                }
+            }
+
+            // wrong complex collection type name
+            responseMessageWithoutModel = new TestStreamResponseMessage(new MemoryStream());
+            responseMessageWithoutModel.SetHeader("Content-Type", mimeType);
+
+            using (var messageWriter = new ODataMessageWriter(responseMessageWithoutModel, settings))
+            {
+                var odataWriter = await messageWriter.CreateODataResourceSetWriterAsync();
+                var complexCollection = new ODataResourceSetWrapper()
+                {
+                    ResourceSet = new ODataResourceSet(),
+                    Resources = new List<ODataResourceWrapper>()
+                        {
+                            TestsHelper.CreatePrimaryContactODataWrapper()
+                        }
+                };
+
+                complexCollection.ResourceSet.SetSerializationInfo(new ODataResourceSerializationInfo()
+                {
+                    ExpectedTypeName = NameSpacePrefix + "ContactDETAIL"
+                });
+
+                await ODataWriterHelper.WriteResourceSetAsync(odataWriter, complexCollection);
+                var result = await TestsHelper.ReadStreamContentAsync(responseMessageWithoutModel.GetStream());
+
+                if (!mimeType.Contains(MimeTypes.ODataParameterNoMetadata))
+                {
+                    // [{"@odata.type":"#Microsoft.Test.OData.Services.AstoriaDefaultService.ContactDetails","EmailBag@odata.type":"#Collection(String)"...
+                    Assert.Contains("\"value\":[{\"@odata.type\":\"#Microsoft.OData.E2E.TestCommon.Common.Server.EndToEnd.ContactDetails\",\"EmailBag", result);
+
+                    // no metadata does not write odata.metadata
+                    Assert.Contains("$metadata#Collection(" + NameSpacePrefix + "ContactDETAIL)", result);
+                }
+                else
+                {
+                    Assert.DoesNotContain("\"@odata.type\":\"#Microsoft.OData.E2E.TestCommon.Common.Server.EndToEnd.ContactDetails\"", result);
+                }
+            }
+        }
+
+        /// <summary>
+        /// Provide EDM model, and then write an entry with wrong serialization info
+        /// </summary>
+        [Theory]
+        [InlineData(MimeTypeODataParameterFullMetadata)]
+        [InlineData(MimeTypeODataParameterMinimalMetadata)]
+        [InlineData(MimeTypeODataParameterNoMetadata)]
+        public async Task WriteEntryWithWrongSerializationInfoWithModel(string mimeType)
+        {
+            bool[] autoComputeMetadataBools = new bool[] { true, false, };
+
+            foreach (var autoComputeMetadata in autoComputeMetadataBools)

```

Comment:
This looks like a good candidate for parameterizing the test further instead of having a loop.